src/kOS.Safe/Execution/CPU.cs

```diff
@@ -614,8 +615,19 @@ private Variable GetVariable(string identifier, bool barewordOkay = false, bool
             }
             if (failOkay)
                 return null;
-            else
-                throw new KOSUndefinedIdentifierException(identifier.TrimStart('$'), "");
+            // In the case where we were looking for a function pointer but didn't find one, and would
+            // have failed with exception, then it's still acceptable to find a hit that isn't a function
+            // pointer (has no trailing asterisk '*') but only if it's a delegate of some sort:
+            if (identifier.EndsWith("*"))
+            {
+                string trimmedTail = identifier.TrimEnd('*');
+                Variable retryVal = GetVariable(trimmedTail, barewordOkay, failOkay);
```

Comment:
This is why I don't always agree with the attitude that you should always change
```
if (foo)
   return something;
else
{
   do something;
}
```
into just
```
if (foo)
   return something;
do something;
```
Even though they have the same effect, there is a conceptual difference in the programmer's mind between those two.  The first form conveys the intent that ``do something`` was intended as being directly married to the ``if (foo)`` check, and thus if you move that if-check at a later date, it should probably move the else clause with it so it stays attached.  The second form divorces the ``do something`` from the ``if (foo)`` check and makes it look as if it should be left behind if there's a reason to move the ``if (foo)`` check later.

In this case, I really see the call to GetVariable as being not directly "else-tied" to the ``if (failOkay)`` check above.  In other words, passing in the variables anyway even though they *happen to be* both false in this case is a form of future-proofing in case sections of his code get moved and shuffled around later.